pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -244,6 +250,40 @@ func (r *ReconcileClusterClaim) Reconcile(request reconcile.Request) (result rec
 	}
 }
 
+func getClaimLifetime(poolLifetime, claimLifetime *metav1.Duration) *metav1.Duration {
+	var lifetime *metav1.Duration
+	switch {
+	case poolLifetime != nil && claimLifetime != nil:
+		lifetime = poolLifetime
+		if claimLifetime.Duration < poolLifetime.Duration {
+			lifetime = claimLifetime
+		}
+	case poolLifetime != nil:
+		lifetime = poolLifetime
+	case claimLifetime != nil:
+		lifetime = claimLifetime
+	}
+	return lifetime
+}
+
+func (r *ReconcileClusterClaim) clusterPoolLifetimeForClaim(claim *hivev1.ClusterClaim, logger log.FieldLogger) (*metav1.Duration, error) {
+	// Fetch the ClusterPool instance
+	clp := &hivev1.ClusterPool{}
+	// claims exists in the same namespace as
+	key := client.ObjectKey{Namespace: claim.Namespace, Name: claim.Spec.ClusterPoolName}
+	err := r.Get(context.TODO(), key, clp)
+	if apierrors.IsNotFound(err) {
+		logger.Info("pool not found")
```

Comment:
Suggest: logger.WithField("pool", key).WithField("claim", claim.Name).Info("cluster pool no longer exists")

Reply:
fixed in 
c29403e